usaspending_api/search/tests/test_spending_by_award_type.py

```diff
@@ -237,3 +237,54 @@ def test_spending_by_award_both_zip_filter(client, mock_matviews_qs):
         }))
     assert len(resp.data['results']) == 1
     assert resp.data['results'][0] == {'internal_id': 1, 'Place of Performance Zip5': '00001'}
+
+
+@pytest.mark.django_db
+def test_spending_by_award_foreign_filter(client, mock_matviews_qs):
+    """ Verify that foreign country filter is returning the correct results """
+    mock_model_1 = MockModel(award_id=1, piid=None, fain='abc', uri=None,
+                             type='B', pulled_from="AWARD", recipient_location_country_name="United States",
+                             recipient_location_country_code="USA")
+    mock_model_2 = MockModel(award_id=2, piid=None, fain='abd', uri=None,
+                             type='B', pulled_from="AWARD", recipient_location_country_name="United States",
+                             recipient_location_country_code="USA")
+    mock_model_3 = MockModel(award_id=3, piid=None, fain='abe', uri=None,
+                             type='B', pulled_from="AWARD", recipient_location_country_name="Gibraltar",
+                             recipient_location_country_code="GIB")
+
+    add_to_mock_objects(mock_matviews_qs, [mock_model_1, mock_model_2, mock_model_3])
+
+    # test simple, single zip
+    resp = client.post(
+        '/api/v2/search/spending_by_award/',
+        content_type='application/json',
+        data=json.dumps({
+            "filters": {"award_type_codes": [
+                        "A",
+                        "B",
+                        "C",
+                        "D"
+                        ]},
+            "fields": ["Award ID"]
+        }))
+    assert len(resp.data['results']) == 3
+
+    # test that adding a zip that has no results doesn't remove the results from the first zip
+    resp = client.post(
+        '/api/v2/search/spending_by_award/',
+        content_type='application/json',
+        data=json.dumps({
+            "filters": {
+                "award_type_codes": [
+                    "A",
+                    "B",
+                    "C",
+                    "D"
+                ],
+                "recipient_scope": "foreign"
+            },
+            "fields": [
+                "Award ID"
+            ],
+        }))
+    assert len(resp.data['results']) == 1
```

Comment:
You should also include an assertion verifying that "recipient_scope: domestic" returns 2 results.

Reply:
Added additional tests that account for country code only, country name only, and country name and country code.